### ISSUE TITLE
Dart Analyzer Error - Encountered a Dart Analysis Server error.

### ISSUE THREAD

**Issue by rockingdice** [dart-lang/sdk#34784]
Analyzer Feedback from IntelliJ

# Version information

- `IDEA IU-182.3911.36`
- `2.1.0-edge.8a919426f0e8665a0e6b01c2fcf3113587f8b040`
- `IU-182.3911.36, JRE 1.8.0_152-release-1248-b8x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.13.6 unknown, screens 1680x1050  Retina`

# Exception

Please append the contents of:
file:///private/var/folders/zf/dtqnb1l96fj0k3lwzpm090qh0000gn/T/report1.txt
```
Dart analysis server, SDK version 2.1.0-edge.8a919426f0e8665a0e6b01c2fcf3113587f8b040, server version 1.20.5, error: Analysis failed: /Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart
Exception: Element not found in summary: package:mhwo/models/page_router_data.dart;package:mhwo/models/page_router_data.g.dart;BuiltList
#0      SummaryResynthesizer.getElement (package:analyzer/src/summary/resynthesize.dart:308:9)
#1      ElementHandle.actualElement (package:analyzer/src/dart/element/handle.dart:345:42)
#2      ClassElementHandle.actualElement (package:analyzer/src/dart/element/handle.dart:35:43)
#3      ClassElementHandle.type (package:analyzer/src/dart/element/handle.dart:111:29)
#4      _ReferenceInfo._buildType (package:analyzer/src/summary/resynthesize.dart:983:24)
#5      _ReferenceInfo.type (package:analyzer/src/summary/resynthesize.dart:929:15)
#6      _ReferenceInfo.buildType (package:analyzer/src/summary/resynthesize.dart:954:15)
#7      _UnitResynthesizer.buildType (package:analyzer/src/summary/resynthesize.dart:1367:28)
#8      SummaryResynthesizerContext.resolveTypeRef (package:analyzer/src/summary/resynthesize.dart:459:30)
#9      ExecutableElementImpl.returnType (package:analyzer/src/dart/element/element.dart:4008:64)
#10     DeclarationResolver.visitMethodDeclaration (package:analyzer/src/generated/declaration_resolver.dart:364:54)
#11     MethodDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:7686:49)
#12     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:8304:20)
#13     ClassDeclarationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:1795:13)
#14     RecursiveAstVisitor.visitClassDeclaration (package:analyzer/dart/ast/visitor.dart:690:10)
#15     DeclarationResolver.visitClassDeclaration.<anonymous closure> (package:analyzer/src/generated/declaration_resolver.dart:89:13)
#16     DeclarationResolver._walk (package:analyzer/src/generated/declaration_resolver.dart:543:13)
#17     DeclarationResolver.visitClassDeclaration (package:analyzer/src/generated/declaration_resolver.dart:88:5)
#18     ClassDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:1765:49)
#19     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:8304:20)
#20     CompilationUnitImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:2437:21)
#21     RecursiveAstVisitor.visitCompilationUnit (package:analyzer/dart/ast/visitor.dart:714:10)
#22     CompilationUnitImpl.accept (package:analyzer/src/dart/ast/ast.dart:2430:49)
#23     DeclarationResolver.resolve (package:analyzer/src/generated/declaration_resolver.dart:51:12)
#24     LibraryAnalyzer._resolveFile (package:analyzer/src/dart/analysis/library_analyzer.dart:576:31)
#25     LibraryAnalyzer._analyze.<anonymous closure> (package:analyzer/src/dart/analysis/library_analyzer.dart:104:9)
#26     __InternalLinkedHashMap&_HashVMBase&MapMixin&_LinkedHashMapMixin.forEach (dart:collection/runtime/libcompact_hash.dart:370:8)
#27     LibraryAnalyzer._analyze (package:analyzer/src/dart/analysis/library_analyzer.dart:103:13)
#28     LibraryAnalyzer.analyze.<anonymous closure> (package:analyzer/src/dart/analysis/library_analyzer.dart:80:14)
<asynchronous suspension>
#29     _PerformanceTagImpl.makeCurrentWhileAsync (package:analyzer/src/generated/utilities_general.dart:222:21)
<asynchronous suspension>
#30     LibraryAnalyzer.analyze (package:analyzer/src/dart/analysis/library_analyzer.dart:79:43)
<asynchronous suspension>
#31     AnalysisDriver._computeAnalysisResult.<anonymous closure> (package:analyzer/src/dart/analysis/driver.dart:1244:71)
<asynchronous suspension>
#32     PerformanceLog.runAsync (package:front_end/src/base/performance_logger.dart:52:21)
<asynchronous suspension>
#33     AnalysisDriver._computeAnalysisResult (package:analyzer/src/dart/analysis/driver.dart:1220:20)
<asynchronous suspension>
#34     AnalysisDriver.performWork (package:analyzer/src/dart/analysis/driver.dart:1025:23)
<asynchronous suspension>
#35     AnalysisDriverScheduler._run (package:analyzer/src/dart/analysis/driver.dart:1916:24)
<asynchronous suspension>
#36     AnalysisDriverScheduler.start (package:analyzer/src/dart/analysis/driver.dart:1857:5)
#37     new AnalysisServer (package:analysis_server/src/analysis_server.dart:397:29)
#38     SocketServer.createAnalysisServer (package:analysis_server/src/socket_server.dart:81:26)
#39     StdioAnalysisServer.serveStdio (package:analysis_server/src/server/stdio_server.dart:37:18)
#40     Driver.start.<anonymous closure> (package:analysis_server/src/server/driver.dart:504:21)
#41     _rootRun (dart:async/zone.dart:1124:13)
#42     _CustomZone.run (dart:async/zone.dart:1021:19)
#43     _runZoned (dart:async/zone.dart:1516:10)
#44     runZoned (dart:async/zone.dart:1463:12)
#45     Driver._captureExceptions (package:analysis_server/src/server/driver.dart:550:12)
#46     Driver.start (package:analysis_server/src/server/driver.dart:502:7)
#47     main (file:///b/build/slave/Mac_Engine/build/src/third_party/dart/pkg/analysis_server/bin/server.dart:12:11)
#48     _AsyncAwaitCompleter.start (dart:async/runtime/libasync_patch.dart:49:6)
#49     main (file:///b/build/slave/Mac_Engine/build/src/third_party/dart/pkg/analysis_server/bin/server.dart:10:10)
#50     _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:287:32)
#51     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)

```

```
1539446394477 => {"id":"90","method":"edit.isPostfixCompletionApplicable","params":{"file":"/Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart","offset":452,"key":".tryon"},"cli
1539446394481 <= {"id":"90","result":{"value":false}}
1539446394481 => {"id":"91","method":"edit.isPostfixCompletionApplicable","params":{"file":"/Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart","offset":452,"key":".par"},"clien
1539446394481 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":false}}}
1539446394482 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":true}}}
1539446394484 <= {"id":"91","result":{"value":false}}
1539446394484 => {"id":"92","method":"edit.isPostfixCompletionApplicable","params":{"file":"/Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart","offset":452,"key":".else"},"clie
1539446394488 <= {"id":"92","result":{"value":false}}
1539446394488 => {"id":"93","method":"edit.isPostfixCompletionApplicable","params":{"file":"/Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart","offset":452,"key":".iter"},"clie
1539446394488 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":false}}}
1539446394489 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":true}}}
1539446394491 <= {"id":"93","result":{"value":false}}
1539446394491 => {"id":"94","method":"edit.isPostfixCompletionApplicable","params":{"file":"/Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart","offset":452,"key":".notnull"},"c
1539446394495 <= {"id":"94","result":{"value":false}}
1539446394495 => {"id":"95","method":"edit.isPostfixCompletionApplicable","params":{"file":"/Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart","offset":452,"key":".fori"},"clie
1539446394495 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":false}}}
1539446394496 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":true}}}
1539446394498 <= {"id":"95","result":{"value":false}}
1539446394498 => {"id":"96","method":"edit.isPostfixCompletionApplicable","params":{"file":"/Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart","offset":452,"key":".try"},"clien
1539446394502 <= {"id":"96","result":{"value":false}}
1539446394502 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":false}}}
1539446394818 => {"id":"97","method":"edit.getAssists","params":{"file":"/Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart","offset":455,"length":0},"clientRequestTime":1539446
1539446394819 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":true}}}
1539446394824 <= {"id":"97","result":{"assists":[]}}
1539446394826 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":false}}}
1539446396509 => {"id":"98","method":"analysis.updateContent","params":{"files":{"/Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart":{"type":"add","content":"import 'package:bu
1539446396510 <= {"id":"98","result":{}}
1539446396510 <= {"event":"server.status","params":{"analysis":{"isAnalyzing":true}}}
1539446396517 => {"id":"99","method":"edit.getAssists","params":{"file":"/Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart","offset":453,"length":0},"clientRequestTime":1539446
1539446396521 <= {"event":"server.error","params":{"isFatal":false,"message":"Analysis failed: /Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart","stackTrace":"Exception: Eleme
```

```
Dart analysis server, SDK version 2.1.0-edge.8a919426f0e8665a0e6b01c2fcf3113587f8b040, server version 1.20.5, error: Analysis failed: /Users/li9s/github/flutter_projects/mhwo/lib/models/page_router_data.dart
Exception: Element not found in summary: package:mhwo/models/page_router_data.dart

**Comment by bwilkerson** [comment]
@scheglov

**Comment by scheglov** [comment]
It was fixed in 765a78d44e7a9e5fd09b10a3b0c60c6e2e4bba96
Try a bleeding edge SDK.
